common/src/main/scala/io/phdata/models/HiveAllocation.scala

```diff
@@ -15,8 +15,19 @@ case class HiveAllocation(
     readonlyGroup: Option[HiveGrant] = None,
     id: Option[Long] = None,
     directoryCreated: Option[Instant] = None,
-    databaseCreated: Option[Instant] = None
-)
+    databaseCreated: Option[Instant] = None,
+    protocol: Option[String] = None
+) {
+
+  def getProtocol: String =
+    if (protocol.isDefined)
+      protocol.get
+    else
+      location.split("://").head match {
+        case "s3a" => "s3a"
+        case _     => "hdfs"
```

Comment:
There can be other protocols not listed here (like adls), so the protocol should just be the substring before the `://`. The above would recognize adls as the hdfs protocol, which isn't right.

Reply:
You can just get rid of the match statement

Reply:
Also if there is no `://` the protocol is HDFS.